src/rev_ai/apiclient.py

```diff
@@ -102,14 +102,27 @@ def submit_job_url(
             Sections of transcript needed to be transcribed.
         :param speaker_names: Only available with "human" transcriber.
             Human readable names of speakers in the file.
+        :param source_config: CustomerUrlData object containing url of the source media and
+            optional authentication headers to use when accessing the source url
+        :param notification_config: CustomerUrlData object containing the callback url to
+            invoke on job completion as a webhook and optional authentication headers to use when
+            calling the callback url
         :returns: raw response data
         :raises: HTTPError
         """
-        if not media_url:
-            raise ValueError('media_url must be provided')
-        payload = self._create_job_options_payload(media_url, metadata,
-                                                   callback_url, skip_diarization,
-                                                   skip_punctuation, speaker_channels_count,
+        if not (media_url or source_config):
+            raise ValueError('media_url or source_config must be provided')
+        check_exclusive_options(media_url, 'media_url', source_config, 'source_config')
```

Comment:
We should rely on the API to validate the model. Don't make extra work for yourself by performing the model validation client side.